coordinator/schedulepod/main.go

```diff
@@ -0,0 +1,790 @@
+	if err != nil {
+		return fmt.Errorf("could not get offline provider: %v", err)
+	}
+	if err != nil {
+		return fmt.Errorf("could not get provider as offline store: %v", err)
+	}
+	liveAddr := fmt.Sprintf("%s:%s", redisHost, redisPort)
+	redisConfig := &provider.RedisConfig{
+		Addr: liveAddr,
+	}
+	serialRedisConfig := redisConfig.Serialized()
+	if err != nil {
+		return fmt.Errorf("could not get provider as online store")
+	}
+	onlineProvider, err := provider.Get(provider.RedisOnline, serialRedisConfig)
+	if err != nil {
+		return fmt.Errorf("could not get online provider")
+	}
+	onlineStore, err := onlineProvider.AsOnlineStore()
+	if err != nil {
+		return fmt.Errorf("could not get provider as online store: %v", err)
+	}
+	featureName := createSafeUUID()
+	sourceName := createSafeUUID()
+	originalTableName := createSafeUUID()
+	if err := CreateOriginalPostgresTable(originalTableName); err != nil {
+		return err
+	}
+	if err := materializeFeatureWithProvider(client, serialPGConfig, serialRedisConfig, featureName, sourceName, originalTableName, "*/1 * * * *"); err != nil {
+		return fmt.Errorf("could not create online feature in metadata: %v", err)
+	}
+	if err := client.SetStatus(context.Background(), metadata.ResourceID{Name: sourceName, Variant: "", Type: metadata.SOURCE_VARIANT}, metadata.READY, ""); err != nil {
+		return err
+	}
+	featureID := metadata.ResourceID{Name: featureName, Variant: "", Type: metadata.FEATURE_VARIANT}
+	featureCreated, err := client.GetFeatureVariant(context.Background(), metadata.NameVariant{Name: featureName, Variant: ""})
+	if err != nil {
+		return fmt.Errorf("could not get feature: %v", err)
+	}
+	if featureCreated.Status() != metadata.CREATED {
+		return fmt.Errorf("Feature not set to created with no coordinator running")
+	}
+	kubeJobSpawner := coordinator.KubernetesJobSpawner{}
+	coord, err := coordinator.NewCoordinator(client, logger, cli, &kubeJobSpawner)
+	if err != nil {
+		return fmt.Errorf("Failed to set up coordinator")
+	}
+	sourceID := metadata.ResourceID{Name: sourceName, Variant: "", Type: metadata.SOURCE_VARIANT}
+	if err := coord.ExecuteJob(metadata.GetJobKey(sourceID)); err != nil {
+		return err
+	}
+	if err := coord.ExecuteJob(metadata.GetJobKey(featureID)); err != nil {
+		return err
+	}
+
+	resourceTable, err := onlineStore.GetTable(featureName, "")
+	if err != nil {
+		return err
+	}
+	for _, record := range testOfflineTableValues {
+		value, err := resourceTable.Get(record.Entity)
+		if err != nil {
+			return err
+		}
+		if !reflect.DeepEqual(value, record.Value) {
+			return fmt.Errorf("Feature value did not materialize")
+		}
+	}
+	featureFirstTimestamp, err := client.GetFeatureVariant(context.Background(), metadata.NameVariant{Name: featureName, Variant: ""})
+	if err != nil {
+		return fmt.Errorf("could not get training set")
+	}
+	oldTimestamp := featureFirstTimestamp.LastUpdated()
+	if err := UpdateOriginalPostgresTable(originalTableName); err != nil {
```

Comment:
Refactor file to change Original to Source or Base. Original isn't entirely clear what this table is